# Analyser les features

In [1]:
import pandas as pd

In [2]:
pd.set_option('display.max_rows', 2)    # Afficher toutes les lignes
pd.set_option('display.max_columns', None) # Afficher toutes les colonnes
pd.set_option('display.width', None)       # Ajuster la largeur de l'affichage

In [3]:
df = pd.read_csv('h5_datas.csv').drop(columns=['genre'])

In [4]:
df

analysis_sample_rate                         audio_md5  danceability  \
0                    22050  94d9ce5ec79b79845b22cb628a4b7b88           0.0   
...                    ...                               ...           ...   
9999                 22050  6b525e31dc172360b16b47f95526950c           0.0   

       duration  end_of_fade_in  energy  idx_bars_confidence  idx_bars_start  \
0     136.77669           0.473     0.0                    0               0   
...         ...             ...     ...                  ...             ...   
9999  243.14730           0.362     0.0                    0               0   

      idx_beats_confidence  idx_beats_start  idx_sections_confidence  \
0                        0                0                        0   
...                    ...              ...                      ...   
9999                     0                0                        0   

      idx_sections_start  idx_segments_confidence  idx_segments_loudness_max  \
0                      0                        0                          0   
...                  ...                      ...                        ...   
9999                   0                        0                          0   

      idx_segments_loudness_max_time  idx_segments_loudness_start  \
0                                  0                            0   
...                              ...                          ...   
9999                               0                            0   

      idx_segments_pitches  idx_segments_start  idx_segments_timbre  \
0                        0                   0                    0   
...                    ...                 ...                  ...   
9999                     0                   0                    0   

      idx_tatums_confidence  idx_tatums_start  key  key_confidence  loudness  \
0                         0                 0    5           0.754    -9.271   
...                     ...               ...  ...             ...       ...   
9999                      0                 0   10           0.146   -21.361   

      mode  mode_confidence  start_of_fade_out    tempo  time_signature  \
0        1            0.532            124.099  155.611               3   
...    ...              ...                ...      ...             ...   
9999     0            0.504            219.220   93.394               4   

      time_signature_confidence            track_id  analyzer_version  \
0                           1.0  TRBBTOM128F42309C2               NaN   
...                         ...                 ...               ...   
9999                        0.0  TRANNLE128E07871BF               NaN   

      artist_7digitalid  artist_familiarity  artist_hotttnesss  \
0                183689            0.074959           0.000000   
...                 ...                 ...                ...   
9999               6208            0.558661           0.300484   

               artist_id  artist_latitude artist_location  artist_longitude  \
0     ARXGVSB11F4C842CF3              NaN             NaN               NaN   
...                  ...              ...             ...               ...   
9999  ARE5OK11187FB450F1         39.73926   Champagne, IL         -89.50409   

                               artist_mbid    artist_name  artist_playmeid  \
0                                      NaN          Opals               -1   
...                                    ...            ...              ...   
9999  02dc2981-7134-4d77-9d3f-7b68e0baced3  Viktor Krauss            45725   

      idx_artist_terms  idx_similar_artists  \
0                    0                    0   
...                ...                  ...   
9999                 0                    0   

                                        release  release_7digitalid  \
0     Doo-Wop Classics Vol. 9 [Beltone Records]              236680   
...                                         ...            

In [5]:
df['song_hotttnesss'].unique()

array([       nan, 0.65362868, 0.62045837, ..., 0.53637802, 0.24610139,
       0.5997103 ])

Parmi la concaténation des 3 groupes de features 'songs':

À ne pas garder:
- analysis_sample_rate => valeur unique: 22050
- audio_md5 => non pertinent
- danceability => valeur unique: 0
- energy => valeur unique: 0
- idx-* => valeur unique: 0
- analyzer_version => valeur unique: 0
- artist 7digitalid => non pertinent
- artist_id => non pertinent
- artist_latitude => non pertinent
- artist_longitude => non pertinent
- artist_mbid => non pertinent
- artist_name => non récupérable depuis l'audio seulement
- artist_playmeid => non pertinent
- release => non récupérable depuis l'audio seulement
- artist_familiarity => non liée à l'audio
- artist_hotttnesss => non liée à l'audio
- artist_location => non liée à l'audio
- song_id => non pertinent
- title => non récupérable à partir de l'audio
- track_7digitalid => non pertinent
- idx_artist_mbtags => non pertinent
- year => non récupérable depuis l'audio

À revoir:
- track_id => pour garder id (à ne pas garder pour le model !)
- song_hotttnesss => c'est  quoi ?

À garder:
- duration
- end_of_fade_in
- key (tonalités)
- key_confidence
- loudness
- mode
- mode_confidence
- start_of_fade_out
- tempo
- time_signature
- time_signature_confidence
- 

TODO: COMMENT EXTRAIRE CES INFORMATIONS (features) D'UN AUDIO (prédiction !) !!!

Feature selection à faire ensuite !!!!

TODO:
Parmis les features présent dans analysis, garder:
